packages/next-mdx/readme.md

```diff
@@ -76,3 +78,74 @@ module.exports = withMDX({
 ## TypeScript
 
 Follow [this guide](https://mdxjs.com/advanced/typescript) from the MDX docs.
+
+---
+
+# App directory
+
+## Installation
+
+For usage with the `app` directory see below.
+
+```
+npm install @next/mdx
+```
+
+or
+
+```
+yarn add @next/md
+```
+
+## Usage
+
+Create a `mdx-component.js` file in your project with the following contents:
+
+```js
+// This file is required to use MDX in the `app` directory.
+export function useMDXComponents(components) {
+  return components
+  // Allows customizing built-in components, e.g. to add styling.
+  // return {
+  //   h1: ({ children }) => <h1 style={{ fontSize: "100px" }}>{children}</h1>,
+  //   ...components,
+  // }
+}
+```
+
+Create a `next.config.js` in your project
+
+```js
+// next.config.js
+const withMDX = require('@next/mdx')({
+  // Optionally provide remark and rehype plugins
+  options: {
+    // If you use remark-gfm, you'll need to use next.config.mjs
+    // as the package is ESM only
+    // https://github.com/remarkjs/remark-gfm#install
+    remarkPlugins: [],
+    rehypePlugins: [],
+    // If you use `MDXProvider`, uncomment the following line.
+    // providerImportSource: "@mdx-js/react",
+  },
+})
+
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  // Configure pageExtensions to include md and mdx
+  pageExtensions: ['ts', 'tsx', 'js', 'jsx', 'md', 'mdx'],
+  experimental: {
+    appDir: true,
+    mdxRs: true
```

Comment:
Is this required?

Reply:
Yeah it is, @next/mdx as JS doesn't use the latest MDX version currently.